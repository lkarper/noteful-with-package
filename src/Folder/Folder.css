.add-note-button {
    border: 1px solid black;
    font-size: 18px;
    text-decoration: none;
    color: black;
    padding: 5px;
}


.add-note-button:hover, .add-note-button:focus {
    background-color: black;
    color: white;
    text-decoration: underline;
}

.delete-folder-button {
    display: block;
    margin: 25px;
}