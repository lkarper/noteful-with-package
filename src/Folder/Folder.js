import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import ShortNote from '../ShortNote/ShortNote';
import NotesContext from '../NotesContext/NotesContext';
import config from '../config';
import './Folder.css';

const Folder = (props) => {

    const [deleteFail, setDeleteFail] = useState(false);

    const deleteFolder = (id, cb) => {
        fetch(`${config.API_URL}/folders/${id}`, {
            method: 'DELETE',
            headers: {
                'content-type': 'application/json',
                'Authorization': `Bearer ${config.API_KEY}`
            },
        })
        .then(res => {
            if (!res.ok) {
              // get the error message from the response,
              return res.json().then(error => {
                // then throw it
                throw error
              })
            }
          })
        .then(data => {
            props.history.push('/');
            cb(id);
        })
        .catch(error => {
            console.log(error);
            setDeleteFail(true);
        });
    }
    return (
        <NotesContext.Consumer>
            {value => {
                const notes = value.notes
                    .filter(note => note.folder === parseInt(props.match.params.folderId))
                    .map(note =>
                        <ShortNote key={note.id} note={note} />
                    );
                
                return (
                    <section className="notes">
                        {!value.notes.length ? <p>Loading...</p> :
                            <>
                                <ul>
                                    {notes}
                                </ul>
                                <Link 
                                    className="add-note-button"
                                    to={{
                                        pathname: "/add-note",
                                        state: {
                                            folderId: props.match.params.folderId,
                                        },
                                    }}
                                >
                                    Add note
                                </Link>
                                <Link 
                                    className="add-note-button"
                                    to={{
                                        pathname: `/update-folder/${props.match.params.folderId}`,
                                        state: {
                                            name: `${value.folders.length ? value.folders.find(f => f.id === parseInt(props.match.params.folderId)).name : null}`,
                                        },
                                    }}
                                >
                                    Change Folder Name
                                </Link>
                                <button className="delete-folder-button" onClick={() => deleteFolder(parseInt(props.match.params.folderId), value.deleteFolder)}>Delete Folder</button>
                            </>
                        }
                        {deleteFail ? <p>Error: failed to delete folder.  Check your connection and try again.</p> : ''}
                    </section>
                );
            }}
        </NotesContext.Consumer>
    );
}

Folder.defaultProps = {
    match: {
        params: {
            folderId: '',
        },
    },
};

Folder.propTypes = {
    match: PropTypes.object.isRequired,
}

export default Folder;