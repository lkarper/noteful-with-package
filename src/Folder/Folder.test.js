import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Folder from './Folder';


describe('Folder component', () => {

    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <Folder />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(
        <BrowserRouter>
            <Folder />
        </BrowserRouter>);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });

});