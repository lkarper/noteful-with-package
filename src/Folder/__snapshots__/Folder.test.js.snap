// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Folder component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Folder
      match={
        Object {
          "params": Object {
            "folderId": "",
          },
        }
      }
    >
      <section
        className="notes"
      >
        <p>
          Loading...
        </p>
      </section>
    </Folder>
  </Router>
</BrowserRouter>
`;
